### PR TITLE
Move to mocha / expect

### PR DESCRIPTION
**Pull request by jnewman**

### CODE REVIEW

File: test/test.html [hunk 1]
```diff
@@ -1,44 +0,0 @@
-<!DOCTYPE html>
```
Comment (jnewman) [inline test/test.html]: Should we spend time making this work w/ the new tests?

File: test/karma-base.config.babel.js [hunk 2]
```diff
@@ -9,36 +13,39 @@ export default {
...
+  },
 
   // generate_expected breaks the path a bit b/c it's writing relative to itself.
   // remap it here to avoid 404s
   proxies: {
     '/remote/': '/base/remote/'
   },
 
+  reporters: [
+    'progress'
```
Comment (sethyates) [inline test/karma-base.config.babel.js]: Not `mocha`?

File: test/karma-base.config.babel.js [hunk 3]
```diff
@@ -47,5 +54,14 @@ export default {
   ],
 
   singleRun: true,
+
+  webpack: Object.assign(webpackConfig, {
```
Comment (sethyates) [inline test/karma-base.config.babel.js]: Just `webpack: webpackConfig`?

File: test/unit/write-comparisons.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import {Html, Js, Uri} from './write-thunks'
```
Comment (sethyates) [inline test/unit/write-comparisons.js]: Move these to a `../fixtures` folder?

File: test/unit/write-comparor.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import _ from 'lodash';
```
Comment (sethyates) [inline test/unit/write-comparor.js]: Move this to a `../helpers` folder?

File: test/unit/write-thunks.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+export class Html {
```
Comment (sethyates) [inline test/unit/write-thunks.js]: Move this to a `../helpers` folder?

File: test/unit/write-transforms.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import _ from 'lodash';
+
+/**
+ * Definitions to swap on the native result. Key is the new string and Value is the pattern to replace
+ *
+ * @type {Object.<string, RegExp>}
+ */
+export const nativeToPostscribe = [_.trim];
```
Comment (sethyates) [inline test/unit/write-transforms.js]: I don't understand what's going on here.  You've declared the type as Object, but it is setup as an array with `_.trim`.

File: test/unit/write-comparor.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+import postscribe from '../../src/postscribe';
+
+import {iframe} from './utils';
+import {Html, Js, Uri} from './write-thunks';
+import {nativeToPostscribe, postscribeToNative} from './write-transforms';
+
+const $ = require('jquery');
+
+export default class WriteComparor {
```
Comment (sethyates) [inline test/unit/write-comparor.js]: No need for this to be a class.  Just export each function individually.

File: test/unit/write-comparor.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        }
+      });
+      const result = doc.body.innerHTML;
+      doc.close();
+      return result;
+    });
+  }
+
+  postscribeResults(...def) {
+    const HELPER_CLASS_NAME = 'ps-writer';
```
Comment (sethyates) [inline test/unit/write-comparor.js]: Can pull this out module level.

File: test/unit/vbscript.spec.js [hunk 10]
```diff
@@ -1,27 +1,28 @@
 /* global $,postscribe,test,ok,start */
```
Comment (sethyates) [inline test/unit/vbscript.spec.js]: Let's get rid of these since all the new stuff should be imported.

File: test/unit/utils.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import Q from 'q';
```
Comment (sethyates) [inline test/unit/utils.js]: Move this out of `units`, to `../helpers`?

File: test/unit/utils.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+import _ from 'lodash';
+
+export const iframe = () => {
```
Comment (sethyates) [inline test/unit/utils.js]: Why not just `export function iframe() {`?

File: test/unit/tokenprocessing.spec.js [hunk 13]
```diff
@@ -92,32 +85,31 @@ $(document).ready(function() {
...
+        beforeWriteToken: tok => {
           if (tok.tagName && tok.tagName.toLowerCase() === 'style') {
             return null;
           }
           return tok;
         },
-        done: function() {
-          ok(div.innerHTML.indexOf('<style') === -1);
-          start();
+        done: () => {
```
Comment (sethyates) [inline test/unit/tokenprocessing.spec.js]: You can probably write this as `done() {`

File: test/unit/tokenprocessing.spec.js [hunk 14]
```diff
@@ -92,32 +85,31 @@ $(document).ready(function() {
...
+    const div = document.createElement('div');
     div.id = 'skip-all-style-declarations';
     document.body.appendChild(div);
-    stop();
     postscribe(div,
       '<style type="text/css">body { background-color: green; }</style>' +
       '<STYLE type="text/css">img { border: 1px solid red; }</STYLE>',
       {
-        beforeWriteToken: function(tok) {
+        beforeWriteToken: tok => {
```
Comment (sethyates) [inline test/unit/tokenprocessing.spec.js]: You can probably write this as `beforeWriteToken(tok) {`

File: test/unit/tokenprocessing.spec.js [hunk 15]
```diff
@@ -1,88 +1,81 @@
...
           if (tok.tagName === type && tok.type !== 'endTag') {
             amount++;
           }
           return tok;
         },
-        done: function() {
-          ok(amount === expectedAmount);
-          start();
+        done: () => {
+          expect(amount === expectedAmount).to.be.ok();
```
Comment (sethyates) [inline test/unit/tokenprocessing.spec.js]: Do they not have `expect(amount).to.eql(expectedAmount)`?

File: test/unit/simplewrites.spec.js [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+import WriteComparor from './write-comparor';
+
+
+describe('Simple writes', function() {
+  const wc = new WriteComparor();
+  const ok = r => expect(r).to.be.ok();
+  const nok = e => expect().fail(e);
+
+  it('empty tag', done => {
+    wc.compare('<span>A<input name="B">C</span>D').then(ok, nok).finally(done);
```
Comment (sethyates) [inline test/unit/simplewrites.spec.js]: You can probably refactor these all using a next level helper, something like:
```
const writes = args => {
  return done => {
    wc.compare(args).then(ok, nok).finally(done);
  };
};

it('empty tag', done => {
    wc.compare('<span>A<input name="B">C</span>D').then(ok, nok).finally(done);
});
```

**Pull request closed without merging**